src/django_idom/types.py

```diff
@@ -51,3 +51,19 @@ class ViewComponentIframe:
     view: View | Callable
     args: Sequence
     kwargs: dict
+
+
+@dataclass
+class QueryOptions:
+    """Configuration options that can be provided to `use_query`."""
+
+    postprocessor_options: dict[str, Any] = field(
+        default_factory=lambda: {"many_to_many": False, "many_to_one": False}
+    )
+    """Configuration values usable by the `postprocessor`."""
+
+    postprocessor: Callable[[_Data, QueryOptions], None] | None = None
+    """A post processing callable that can read/modify the query `data` and the `QueryOptions` object.
+
+    If unset, the default handler is used. This handler can be configured via `postprocessor_options`
+    to recursively fetch all fields to ensure queries are not performed lazily."""
```

Comment:
Yes, for example sometimes it's more convenient for me to always have the data in json format rather than as a Django `QuerySet`, so I would likely use a generic postprocessor that serializes the data.

IMO it's prettier to have this built-in to the hook.

We'll have a chance to rework this API to be more akin to `React Query` when we migrate this hook into core.

Reply:
> IMO it's prettier to have this built-in to the hook.

This may be the case if `postprocessor` defaults to prefetching, but it has the con that the default options no longer make sense if you say, set `postprocessor=to_json`. Also, if I mistakenly configure `postprocessor_options` while using the `to_json` post processor:

```python
QueryOptions(postprocessor_options={"many_to_many": True}, postprocessor=to_json)
```

 I won't be notified that my `postprocessor_options` will have no effect because `to_json` simply won't interpret them.

Reply:
Maybe the solution here is to convert `postprocessor_options` into kwargs for `postprocessor`.

That way, an exception will be generated for incorrect options.